lib/aoi.js/src/functions/discord/author/$username.ts

```diff
@@ -0,0 +1,75 @@
+import FunctionBuilder from '@aoi.js/core/builders/Function.js';
+import { parseString } from '@aoi.js/core/parsers/string.js';
+import { ErrorCode, FunctionType, ReturnType } from '@aoi.js/typings/enum.js';
+import { escapeResult } from '@aoi.js/utils/Helpers/core.js';
+import AoiError from '@aoi.js/core/Error.js';
+
+/**
+ * Returns the username of the user.
+ * @example
+ * ```aoi
+ * ---
+ * name: username
+ * type: basic
+ * ---
+ *
+ * $username // returns the username of the current user
+ * $username[userid] // returns the username of the specified user
+ * ```
+ */
+
+const $username = new FunctionBuilder()
+	.setName('$username')
+	.setType(FunctionType.Getter)
+	.setReturns(ReturnType.String | ReturnType.Void)
+	.setBrackets(true)
+	.setOptional(true)
+	.setFields([
+		{
+			name: 'userId',
+			type: ReturnType.String,
+			required: false,
+			description: 'UserId to search for',
+		},
+	])
+	.setCode((data, scopes, thisArg) => {
+		const [userId] = thisArg.getParams(data);
+
+		let res;
+		if (!userId) {
+			res = thisArg.getResultString(
+				(discordData) => discordData.author?.username,
+			);
+			const escaped = escapeResult(res);
+			return {
+				code: escaped,
+				scope: scopes,
+			};
```

Comment:
don't do this return here

you can use if and else to set value of res and keep this part outside of if/else